src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -106,22 +106,31 @@
             transaction {:id         id
                          :from       from
                          :to         to
+                         :to-name    (when (= to constants/contract-address)
+                                       (i18n/label :t/new-contract))
                          :symbol     symbol
                          :value      (money/bignumber (or value 0))
                          :data       data
-                         :gas        (money/bignumber (money/to-decimal gas))
-                         :gas-price  (money/bignumber (money/to-decimal gasPrice))
+                         :gas        (when (seq gas)
+                                       (money/bignumber (money/to-decimal gas)))
+                         :gas-price  (when (seq gasPrice)
+                                       (money/bignumber (money/to-decimal gasPrice)))
                          :timestamp  now
                          :message-id message_id}
-            sending-from-chat? (not (get-in db [:wallet :send-transaction :waiting-signal?]))
+            sending-from-bot-or-dapp? (not (get-in db [:wallet :send-transaction :waiting-signal?]))
             new-db (assoc-in db [:wallet :transactions-unsigned id] transaction)
             sending-db {:id         id
-                        :from-chat? sending-from-chat?}]
-        (if sending-from-chat?
-          ;;SENDING FROM CHAT
-          {:db       (assoc-in new-db [:wallet :send-transaction] sending-db) ; we need to completely reset sending state here
-           :dispatch-n [[:update-wallet (map :symbol (tokens/tokens-for (ethereum/network->chain-keyword (:network db))))] [:navigate-to-modal :wallet-send-transaction-modal]]}
-          ;;SEND SCREEN WAITING SIGNAL
+                        :from-chat? sending-from-bot-or-dapp?}]
+        (if sending-from-bot-or-dapp?
+          ;;SENDING FROM BOT (CHAT) OR DAPP
+          {:db         (assoc-in new-db [:wallet :send-transaction] sending-db) ; we need to completely reset sending state here
+           :dispatch-n [[:update-wallet (map :symbol (tokens/tokens-for (ethereum/network->chain-keyword (:network db))))]
+                        [:navigate-to-modal :wallet-send-transaction-modal]
+                        (when-not (seq gas)
```

Comment:
Also why seq here? Is `gas` supposed to be a sequence?

Reply:
`string` or `nil`

Reply:
Then `when` should be sufficient? Or you want to filter the empty string too?

Reply:
yes

Reply:
Maybe `clojure.string/blank?` is clearer? And a little more efficient :)

Reply:
tbh i used it before, but now i like more `seq` :) and as i know most devs prefer `seq`

Reply:
`(when (seq s))` is idiomatic to filter empty sequence (as opposed to `(when-not (empty? s))`) but I don't think it is for string? 
Not very important, but something we should consider for our style guide.

Reply:
@jeluard `blank?` is not exactly the opposite of `seq`, an only whitespaces string is `blank?` but it is a valid `seq`.

`seq` is also likely more efficient, it returns a lazy seq, does not evaluate the whole thing, while blank might do. 

In this case if we want to filter out stuff like `"\n"` or `"   "`, blank is the only choice. Otherwise I guess it's a matter of style.

Reply:
correction actually, `seq` might return a lazy seq, depends on the input.

Reply:
so yes, actually you are right, this is unlikely a lazy seq, so blank? should be faster my bad!

Reply:
Ok, last update, sorry I got down the rabbit hole:
in clojurescript (node) with a non-empty string "aa..." (4000000  characters)
```
TIMED BLANK                                                                                                                                                                                    
"Elapsed time: 63.261823 msecs"                                                                                                                                                                
TIMED SEQ                                                                                                                                                                                      
"Elapsed time: 1.657652 msecs"   
```
looking at the code, it depends on the data structure, it returns an iterator when possible, so might be O(1), seems `seq` performs better on strings, probably because it implements https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols 

ok, sorry I stop now pestering people now :), `blank?` seems more appropriate in this case btw

Reply:
Nice catch! Looks like the underlying closure function does a costly copy before making the actual empty check ..